homeassistant/components/signal_messenger/notify.py

```diff
@@ -43,37 +54,88 @@ def get_service(hass, config, discovery_info=None):
 class SignalNotificationService(BaseNotificationService):
     """Implement the notification service for SignalMessenger."""
 
-    def __init__(self, recp_nrs, signal_cli_rest_api):
+    def __init__(
+        self,
+        recp_nrs: list[str],
+        signal_cli_rest_api: SignalCliRestApi,
+    ) -> None:
         """Initialize the service."""
 
         self._recp_nrs = recp_nrs
         self._signal_cli_rest_api = signal_cli_rest_api
 
-    def send_message(self, message="", **kwargs):
-        """Send a message to a one or more recipients.
-
-        Additionally a file can be attached.
-        """
+    def send_message(self, message: str = "", **kwargs: Any) -> None:
+        """Send a message to a one or more recipients. Additionally a file can be attached."""
 
         _LOGGER.debug("Sending signal message")
 
         data = kwargs.get(ATTR_DATA)
 
-        filenames = None
-        if data is not None:
-            if ATTR_FILENAMES in data:
-                filenames = data[ATTR_FILENAMES]
-            if ATTR_FILENAME in data:
-                _LOGGER.warning(
-                    "The 'attachment' option is deprecated, please replace it with 'attachments'. This option will become invalid in version 0.108"
-                )
-                if filenames is None:
-                    filenames = [data[ATTR_FILENAME]]
-                else:
-                    filenames.append(data[ATTR_FILENAME])
+        filenames = self.get_filenames(data)
+        attachments_as_bytes = self.get_attachments_as_bytes(
+            data, CONF_MAX_ALLOWED_DOWNLOAD_SIZE_BYTES
+        )
 
         try:
-            self._signal_cli_rest_api.send_message(message, self._recp_nrs, filenames)
+            self._signal_cli_rest_api.send_message(
+                message, self._recp_nrs, filenames, attachments_as_bytes
+            )
         except SignalCliRestApiError as ex:
             _LOGGER.error("%s", ex)
             raise ex
+
+    @staticmethod
+    def get_filenames(data: Any) -> list[str] | None:
+        """Extract attachment filenames from data."""
+        if data is None:
+            return None
+
+        if ATTR_FILENAMES in data:
+            if isinstance(data[ATTR_FILENAMES], list):
+                return data[ATTR_FILENAMES]
+
+            raise ValueError(f"'{ATTR_FILENAMES}' property must be a list")
+
+        return None
+
+    @staticmethod
+    def get_attachments_as_bytes(
+        data: Any, attachment_size_limit: int
+    ) -> list[bytearray] | None:
+        """Retrieve attachments from URLs defined in data."""
+        if data is None or ATTR_URLS not in data:
+            return None
+
+        if not isinstance(data[ATTR_URLS], list):
+            raise ValueError(f"'{ATTR_URLS}' property must be a list")
+
+        attachments_as_bytes: list[bytearray] = []
+
+        urls = data[ATTR_URLS]
+        for url in urls:
+            try:
+                resp = requests.get(url, verify=False, timeout=10, stream=True)
```

Comment:
Why is verify set to off? I see some people may want it off, but it can pose a security risk for some. Can this instead be a config option with it defaulted to on?

Reply:
That's a good idea - the particular use case we had was downloading files from a camera's local API, so the connection was not https. Making this optional makes more good sense.

Reply:
If it's a local camera, it should be using http? Turning verify off only disables ssl certificate verification.

Reply:
This is now made optional in a25bbc70353e0aad07b3e769e58794e6d682748a. Docs have been updated to show the new attribute